File: backend/app/schemas/drift.py
"""Drift API 요청/응답 스키마.

설계서: 2026-06-14-decision-truth-loop-design.md §5.5
"""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.drift import DriftStatus, DriftType


class DriftOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    type: DriftType
    status: DriftStatus
    branch: str
    external_id: str | None
    detail: str
    opened_at: datetime
    resolved_at: datetime | None


class DriftListOut(BaseModel):
    items: list[DriftOut]
    total: int


class DriftPatchIn(BaseModel):
    action: str   # "ignore" | "reopen"

File: frontend/src/types/drift.ts
// Backend DriftType / DriftStatus enum wire values
export type DriftType = 'decision_not_promoted' | 'status_contradiction';
export type DriftStatus = 'open' | 'resolved' | 'ignored';

// PATCH action
export type DriftAction = 'ignore' | 'reopen';

export interface DriftSummary {
  id: string;
  type: DriftType;
  status: DriftStatus;
  branch: string;
  external_id: string | null;
  detail: string;
  opened_at: string;          // ISO datetime
  resolved_at: string | null;
}

export interface DriftListResponse {
  items: DriftSummary[];
  total: number;
}

export interface DriftStatusUpdateRequest {
  action: DriftAction;
}
